lib/analytics/tracks.ts

```diff
@@ -1,50 +1,87 @@
+import * as T from '../types';
+
+declare global {
+  interface Window {
+    wpcom: {
+      _tkq: TKQItem[];
+      tracks: TracksAPI;
+    };
+  }
+}
+
+declare const TRACKS_COOKIE_DOMAIN: string | undefined;
+
+export type TKQItem = Function | [keyof TracksAPI, string, T.JSONValue];
+
+export type TracksAPI = {
+  storeContext: (c: object) => void;
+  identifyUser: (user: string, login: string) => void;
+  recordEvent: (name: string, props: T.JSONSerializable) => void;
+  setProperties: (properties: object) => void;
+  clearIdentity: () => void;
+};
+
+type Query = Partial<{
+  _dl: string;
+  _dr: string;
+  _en: string;
+  _ht: number;
+  _lg: string;
+  _pf: string;
+  _sx: number;
+  _sy: number;
+  _ts: number;
+  _tz: number;
+  _ui: string | null;
+  _ul: string;
+  _ut: string;
+  _wd: number;
+  anonId: string;
+}>;
+
 window.wpcom = window.wpcom || {};
 window._tkq = window._tkq || [];
 
 window.wpcom.tracks = (function() {
-  var userId,
-    userIdType,
-    userLogin,
-    localCache = {},
-    context = {},
-    pixel = 'https://pixel.wp.com/t.gif',
-    cookieDomain = null,
-    cookiePrefix = 'tk_',
-    testCookie = 'tc',
-    userNameCookie = 'ni',
-    userAnonCookie = 'ai',
-    queriesCookie = 'qs',
-    queriesTTL = 1800,
-    queriesPending = {};
-
-  var getCookie = function(key) {
-    var name =
-        cookiePrefix + encodeURIComponent(key).replace(/[-.+*]/g, '\\$&'),
-      pattern = new RegExp(
-        '(?:(?:^|.*;)\\s*' + name + '\\s*\\=\\s*([^;]*).*$)|^.*$'
-      );
+  let userId: string | null | undefined;
+  let userIdType: string;
+  let userLogin: string | null | undefined;
+  const localCache: { [key: string]: string } = {};
+  let context = {};
+  let pixel = 'https://pixel.wp.com/t.gif';
+  let cookieDomain: string | null = null;
+  const cookiePrefix = 'tk_';
+  const testCookie = 'tc';
+  const userNameCookie = 'ni';
+  const userAnonCookie = 'ai';
+  const queriesCookie = 'qs';
+  const queriesTTL = 1800;
+  const queriesPending: { [key: string]: boolean } = {};
+
+  const getCookie = function(key: string) {
+    const name = `${cookiePrefix}${encodeURIComponent(key).replace(
+      /[-.+*]/g,
+      '\\$&'
+    )}`;
+    const pattern = new RegExp(
+      `(?:(?:^|.*;)\\s*${name}\\s*\\=\\s*([^;]*).*$)|^.*$`
+    );
     return decodeURIComponent(document.cookie.replace(pattern, '$1')) || null;
   };
 
-  var checkCookieDomain = function(domain) {
-    var time = new Date().getTime();
-    document.cookie =
-      cookiePrefix +
-      testCookie +
-      '=' +
-      time +
-      '; domain=' +
-      domain +
-      '; path=/;';
+  const checkCookieDomain = function(domain: string) {
+    const time = new Date().getTime();
+    document.cookie = `${cookiePrefix}${testCookie}=${time}; domain=${domain}; path=/;`;
+    // @ts-ignore
```

Comment:
this code includes an error that the type-checker caught but I don't want to change the behavior in attempting to fix it, accidentally breaking something. I figured that in time we can search for `@ts-ignore` for cleanup work if we want to swing back on this, probably even replacing the analytics module at some point.

That is: I wanted to limit the scope of this patch to preserve existing behaviors even if they are "wrong"

The condition must always return false because it triple-equality-check compares a string and a number.